src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -617,24 +629,48 @@ private Node createToolbar() {
+        rightSideButtons.put("7", factory.createIconButton(StandardActions.PASTE, new OldDatabaseCommandWrapper(Actions.PASTE, this, Globals.stateManager)));
+        rightSideButtons.put("8", factory.createIconButton(StandardActions.CLEANUP_ENTRIES, new OldDatabaseCommandWrapper(Actions.CLEANUP, this, Globals.stateManager)));
+        rightSideButtons.put("9", factory.createIconButton(pushToExternal.getMenuAction(), pushToExternal));
+        rightSideButtons.put("10", factory.createIconButton(StandardActions.FORK_ME, new OpenBrowserAction("https://github.com/JabRef/jabref")));
+        rightSideButtons.put("11", factory.createIconButton(StandardActions.OPEN_FACEBOOK, new OpenBrowserAction("https://www.facebook.com/JabRef/")));
+        rightSideButtons.put("12", factory.createIconButton(StandardActions.OPEN_TWITTER, new OpenBrowserAction("https://twitter.com/jabref_org")));
+
+        HBox rightSide = new HBox(rightSideButtons.values().toArray(new Button[12])
+//                factory.createIconButton(StandardActions.NEW_ENTRY, new NewEntryAction(this, BiblatexEntryTypes.ARTICLE, dialogService, Globals.prefs)),
+//                factory.createIconButton(StandardActions.DELETE_ENTRY, new OldDatabaseCommandWrapper(Actions.DELETE, this, Globals.stateManager)),
+
+//                factory.createIconButton(StandardActions.UNDO, new OldDatabaseCommandWrapper(Actions.UNDO, this, Globals.stateManager)),
+//                factory.createIconButton(StandardActions.REDO, new OldDatabaseCommandWrapper(Actions.REDO, this, Globals.stateManager)),
+//                factory.createIconButton(StandardActions.CUT, new OldDatabaseCommandWrapper(Actions.CUT, this, Globals.stateManager)),
+//                factory.createIconButton(StandardActions.COPY, new OldDatabaseCommandWrapper(Actions.COPY, this, Globals.stateManager)),
+//                factory.createIconButton(StandardActions.PASTE, new OldDatabaseCommandWrapper(Actions.PASTE, this, Globals.stateManager)),
+
+//                factory.createIconButton(StandardActions.CLEANUP_ENTRIES, new OldDatabaseCommandWrapper(Actions.CLEANUP, this, Globals.stateManager)),
+//                factory.createIconButton(pushToExternal.getMenuAction(), pushToExternal),
+
+//                factory.createIconButton(StandardActions.FORK_ME, new OpenBrowserAction("https://github.com/JabRef/jabref")),
+//                factory.createIconButton(StandardActions.OPEN_FACEBOOK, new OpenBrowserAction("https://www.facebook.com/JabRef/")),
+//                factory.createIconButton(StandardActions.OPEN_TWITTER, new OpenBrowserAction("https://twitter.com/jabref_org"))
         );
 
+        uiUpdate.addObserver((o, arg) -> {
+            DefaultTaskExecutor.runInJavaFXThread(() -> {
+                rightSide.getChildren().clear();
+                PushToApplicationButton pushExternal = new PushToApplicationButton(this, pushApplications.getApplications());
```

Comment:
I like that you try to update the ui when the preferences is changed. However, this seems to be a bit too complicated here. We have to probably think about how to make the preference values observable on a global scale (not specific to the pull to app button). But this is a huge project and not in the scope of this PR. Hence, I'd favor the solution that the push to app button is only updated when JabRef is started and adding a short notification that JabRef needs to be restarted for the changes in the prefs to come into effect (have a look at the other prefs panes for such a text).

Reply:
Ok, that sounds reasonable. I will tackle this soon. should be simple to implement.